src/Data/ExistsRE.purs

```diff
@@ -0,0 +1,20 @@
+module Data.ExistsRE where
+
+import Unsafe.Coerce (unsafeCoerce)
+
+-- | A variant of `Exists` that works for type constructors that accept a _row_
+-- | of effects.
+newtype ExistsRE (f :: # ! -> *) = ExistsRE (forall eff. f eff)
```

Comment:
Yeah, that's the best we can do I think. Another option would be to make it foreign explicitly, do you think that would better express what's going on? (In an "all bets are off" kind of way)